popup/popup.js

```diff
@@ -11,14 +11,24 @@ window.onload = () => {
   }
 
   var radios = document.forms['time-form'].elements['time'];
+  // Reference for the timeout so we can clear it midway through
+  let timeoutHandle; 
   for (var i = 0, max = radios.length; i < max; i++) {
     radios[i].onclick = function () {
+      if (timeoutHandle) // reset remaining time 
+          window.clearTimeout(timeoutHandle);
+
       let time = this.value;
+      const statusText = document.getElementById('status');
```

Comment:
@jcads I think L#22 could be outside the for a loop as this is independent of the for-loop.